install/kubernetes/cilium/charts/config/templates/cni-configmap.yaml

```diff
@@ -0,0 +1,35 @@
+{{- $defaultCniVersion := "0.3.1" -}}
+{{- $defaultCniName := "cilium" -}}
+{{- $defaultCniType := "cilium-cni" -}}
+{{- $defaultCniDebug := false -}}
+{{- $defaultCniFirstInterfaceIndex := 1 -}}
+
+{{- if .Values.global.cni.managedConfigmap }}
+# if we want to customize CNI config we can do this in the following path: .global.cni.* after we define .global.cni.managedConfigmap
+# supported config params as of now:
+# .global.cni.version
+# .global.cni.name
+# .global.cni.type
+# .global.cni.debug
+# .global.cni.eni.first_interface_index
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cilium-cni-configuration
+  namespace: {{ .Release.Namespace }}
+data:
+  cni-config: |
+    {
+      "cniVersion": {{ .Values.global.cni.version | default $defaultCniVersion }},
+      "name": {{ .Values.global.cni.name | default $defaultCniName }},
+      "type": {{ .Values.global.cni.type | default $defaultCniType }},
+      "enable-debug": {{ .Values.global.cni.debug | default $defaultCniDebug }},
+{{- if .Values.global.cni.eni }}
```

Comment:
Bringing back the discussion from previous PR https://github.com/cilium/cilium/pull/12897#discussion_r470745214:

> > So far, there has been no option global.cni.eni. The EKS/ENI docs use global.eni. Is this what you meant?

> partially.
> My goal here was to configure CNI to set index of first used ENI.
> Potentially there may be params other than first-interface-index, so I used global.cni.eni. as placeholder. And we > will be able to use something like this if needed:
> 
> ```
> global:
>   cni
> :
>     eni:
>       first-interface-index: 0
>       blablaparam: 111
>    ```

> No option before - not breaking change.

I think the first-interface index portion seems fine.

The piece that I'm wondering about is that we do not currently direct users to specify `--set global.cni.eni=...`. Additionally it seems confusing to have both `--set global.eni=...` as well as `--set global.cni.eni=...`. Does this need to be governed by a new configuration option?

Reply:
yes, you do not direct. So it produces incorrect cluster setup which should be addressed from one side or another. Cilium or EKS (in my case). We will not change EKS part here, Chaining looks like less tested/trusted part here (I was told so). And I like chaining less. So for us the best approach for now - change CNI config with this configmap.

After this PR documentation should be changed. If it would be in this repo - I would update it as well. (personally I often check code/charts to see how documentation differs from reality, so natural for me)

With `--set global.eni=...` I think you enable/disable ENI itself.
With `--set global.cni.eni=` you configure CNI and some part of this CNI config related to ENI. Personally I don't see this confusing. For me it's logical and structured. 

If you can think about better option - let's do this. I don't like hardcoding separate params so I made a place to cover the whole CNI config in one tree.

Reply:
> So for me (without this PR and manual configmap) chart produces incorrect cluster setup which should be addressed from one side or another (Cilium or EKS in my case). I believe that silently limiting amount of allocatable pods per node makes whole setup incorrect, but I got reasons why you do this.

I'm not sure if there was a previous discusson on this point, but I'm not personally very familiar with the ENI side of things. I may /cc in @tgraf / @ungureanuvladvictor / @christarazi in case they may have any feedback here.

Overall though your argument sounds reasonable to me. I also realise that `--set global.eni` is already directly a value rather than an object, so we couldn't change it to have subfields without potentially breaking some things.

One more comment however: I don't think that these options need to be global, do they? I would suggest specifying them as `--set config.cni.eni=...` instead. Globals in general seem to cause more trouble than we realised when they were first introduced in Cilium's Helm charts and we're trying to avoid using them if it's possible.